cpp/cuproj/CMakeLists.txt

```diff
@@ -0,0 +1,276 @@
+
+rapids_cmake_support_conda_env(conda_env MODIFY_PREFIX_PATH)
+
+###################################################################################################
+# - compiler options ------------------------------------------------------------------------------
+
+rapids_cuda_init_runtime(USE_STATIC ${CUDA_STATIC_RUNTIME})
+
+rapids_find_package(
+  CUDAToolkit REQUIRED
+  BUILD_EXPORT_SET cuproj-exports
+  INSTALL_EXPORT_SET cuproj-exports
+)
+# * find CUDAToolkit package
+# * determine GPU architectures
+# * enable the CMake CUDA language
+# * set other CUDA compilation flags
+include(cmake/modules/ConfigureCUDA.cmake)
+
+###################################################################################################
+# - dependencies ----------------------------------------------------------------------------------
+
+# add third party dependencies using CPM
+rapids_cpm_init()
+# find or add cuDF
+include(cmake/thirdparty/CUPROJ_GetCUDF.cmake)
+# find or install GoogleTest and Proj
+if (CUPROJ_BUILD_TESTS)
+    include(cmake/thirdparty/get_gtest.cmake)
+    include(cmake/thirdparty/get_proj.cmake)
+endif()
+
+###################################################################################################
+# - library targets -------------------------------------------------------------------------------
+
+add_library(cuproj
+    src/test.cu
+)
+
+set_target_properties(cuproj
+    PROPERTIES BUILD_RPATH                         "\$ORIGIN"
+               INSTALL_RPATH                       "\$ORIGIN"
+               # set target compile options
+               CXX_STANDARD                        17
+               CXX_STANDARD_REQUIRED               ON
+               CUDA_STANDARD                       17
+               CUDA_STANDARD_REQUIRED              ON
+               POSITION_INDEPENDENT_CODE           ON
+               INTERFACE_POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_options(cuproj
+            PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_FLAGS}>"
+                    "$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_FLAGS}>"
+)
+
+target_compile_definitions(cuproj
+            PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_DEFINITIONS}>"
+                   "$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_DEFINITIONS}>>"
+)
+
+# Specify include paths for the current target and dependents
+target_include_directories(cuproj
+           PUBLIC      "$<BUILD_INTERFACE:${CUPROJ_SOURCE_DIR}/include>"
+           PRIVATE     "$<BUILD_INTERFACE:${CUPROJ_SOURCE_DIR}/src>"
+           INTERFACE   "$<INSTALL_INTERFACE:include>")
+
+# Add Conda library and include paths if specified
+if(TARGET conda_env)
+    target_link_libraries(cuproj PRIVATE "$<BUILD_INTERFACE:conda_env>")
+endif()
+
+# Per-thread default stream
+if(PER_THREAD_DEFAULT_STREAM)
+    target_compile_definitions(cuproj PUBLIC CUDA_API_PER_THREAD_DEFAULT_STREAM)
```

Comment:
should this be public?

Reply:
It's not private in cuDF, RMM or cuSpatial...